launch/launch/actions/execute_process.py

```diff
@@ -1,885 +1,318 @@
-            self.__shutdown_future = create_future(context.asyncio_loop)
-            self.__expand_substitutions(context)
-            self.__logger = launch.logging.get_logger(self.__name)
-            self.__stdout_logger, self.__stderr_logger = \
-                launch.logging.get_output_loggers(self.__name, self.__output)
-            context.asyncio_loop.create_task(self.__execute_process(context))
-        except Exception:
-            for event_handler in event_handlers:
-                context.unregister_event_handler(event_handler)
-            raise
-        return None
-
-    def get_asyncio_future(self) -> Optional[asyncio.Future]:
-        """Return an asyncio Future, used to let the launch system know when we're done."""
-        return self.__completed_future
-
-    @property
-    def name(self):
-        """Getter for name."""
-        return self.__name
-
-    @property
-    def cmd(self):
-        """Getter for cmd."""
-        return self.__cmd
-
-    @property
-    def cwd(self):
-        """Getter for cwd."""
-        return self.__cwd
-
-    @property
-    def env(self):
-        """Getter for env."""
-        return self.__env
-
-    @property
-    def additional_env(self):
-        """Getter for additional_env."""
-        return self.__additional_env
-
-    @property
-    def shell(self):
-        """Getter for shell."""
-        return self.__shell
-
-    @property
-    def prefix(self):
-        """Getter for prefix."""
-        return self.__prefix
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the ExecuteProcess action."""
+
+import shlex
+import threading
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Text
+
+from .execute_local import ExecuteLocal
+
+from ..descriptions import Executable
+from ..frontend import Entity
+from ..frontend import expose_action
+from ..frontend import Parser
+from ..some_substitutions_type import SomeSubstitutionsType
+from ..substitutions import TextSubstitution
+
+_global_process_counter_lock = threading.Lock()
+_global_process_counter = 0  # in Python3, this number is unbounded (no rollover)
+
+
+@expose_action('executable')
+class ExecuteProcess(ExecuteLocal):
+    """Action that begins executing a process and sets up event handlers for the process."""
+
+    def __init__(
+            self,

```

Comment:
@roger-strain @mlanting indentation here seems to have been doubled. I wonder why the linter did not complain.

Reply:
I think it's because these are function arguments rather than the function code.